packages/edit-post/src/hooks/commands/use-common-commands.js

```diff
@@ -131,6 +137,9 @@ export default function useCommonCommands() {
 		label: __( 'Toggle top toolbar' ),
 		callback: ( { close } ) => {
 			toggle( 'core/edit-post', 'fixedToolbar' );
+			if ( isDistractionFree ) {
```

Comment:
For now this is the only one left from the issue, because we call directly the preferences store action.